src/main/java/org/broadinstitute/consent/http/db/mapper/DacReducer.java

```diff
@@ -0,0 +1,83 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.Map;
+import java.util.Objects;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.models.Dataset;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.jdbi.v3.core.mapper.MappingException;
+import org.jdbi.v3.core.result.LinkedHashMapRowReducer;
+import org.jdbi.v3.core.result.RowView;
+
+public class DacReducer implements LinkedHashMapRowReducer<Integer, Dac>,
+    ConsentLogger {
+
+  private final DataUseParser dataUseParser = new DataUseParser();
+
+  @Override
+  public void accumulate(Map<Integer, Dac> container, RowView rowView) {
+
+    try {
+
+      Dac dac =
+          container.computeIfAbsent(
+              rowView.getColumn("dac_id", Integer.class), id -> rowView.getRow(Dac.class));
+
+      if (Objects.nonNull(rowView.getColumn("daa_daa_id", Integer.class))) {
+        DataAccessAgreement daa = new DataAccessAgreement();
+
+
+        if (rowView.getColumn("daa_daa_id", Integer.class) != null) {
+          daa = rowView.getRow(DataAccessAgreement.class);
+        }
+
+
+        if (rowView.getColumn("file_storage_object_id", String.class) != null) {
+          FileStorageObject fso = rowView.getRow(FileStorageObject.class);
+          daa.setFile(fso);
+        }
+
+        if (daa.getDaaId() != null) {
+          dac.setAssociatedDaa(daa);
+        }
+
+      }
+
+      if (Objects.nonNull(rowView.getColumn("dataset_id", Integer.class))) {
+        Dataset dataset = rowView.getRow(Dataset.class);
+
+        //aliased columns must be set directly
+        String dsAlias = rowView.getColumn("dataset_alias", String.class);
+        if (dsAlias != null) {
+            dataset.setAlias(Integer.parseInt(dsAlias));
+        }
```

Comment:
The original code had a try around the integer parsing which was necessary in the past due to invalid dev data. We should try and maintain that condition here:
```
          if (dsAlias != null) {
            try {
              dataset.setAlias(Integer.parseInt(dsAlias));
            } catch (Exception e) {
              logException("Exception parsing dataset alias: " + dsAlias, e);
            }
          }
```